lib/bloc/auth/auth_bloc.dart

```diff
@@ -79,5 +79,17 @@ class AuthBloc extends Bloc<AuthEvent, AuthState> {
       await event.user.sendEmailVerification();
       yield AuthEmailResent(event.user);
     }
+    if (event is UpdatePassword) {
+      var user = _authRepository.loggedInUser();
+      // TODO: Validate old password here
+      await user.updatePassword(event.newPassword);
+
+      yield AuthInitial();
+    }
+    if (event is UpdateName) {
+      var user = _authRepository.loggedInUser();
+      await user.updateProfile(displayName: event.name);
```

Comment:
This does not update the username for the backend. You'd need to update the user model from the backend.